packages/authorization/src/types.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/authorization
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingAddress, Context} from '@loopback/context';
+
+/**
+ * Voting decision for the authorization decision
+ */
+export enum VotingDecision {
+  ALLOW = 'ALLOW',
+  DENY = 'DENY',
+  ABSTAIN = 'ABSTAIN',
+}
+
+/**
+ * A voter function
+ */
+export interface Voter {
+  (ctx: Context): Promise<VotingDecision>;
+}
+
+export const EVERYONE = '$everyone';
+export const AUTHENTICATED = '$authenticated';
```

Comment:
@raymondfeng , the authorization component checks with the authentication component if someone is properly authenticated?

Reply:
Yes, there will be authentication information available in the context that authorization depends on.

- authentication - validates who is requesting access
- authorization - decides if the requestor can access the target resource